source/auth/tests/mongodb-oidc.rst

```diff
@@ -61,37 +58,91 @@ Prose Tests
   handshake, and again during reauthentication).
 - Close the client.
 
-3. Callback is called twice on handshake authentication failure
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+3. Authentication failures with cached tokens retry with a new token
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+- Create a ``MongoClient`` configured with ``retryReads=false`` and a custom
+  OIDC callback that implements the AWS provider logic.
+- Set a fail point for ``find`` commands of the form:
+
+.. code:: javascript
 
-- Create a ``MongoClient`` configured with an ``OIDCCallback`` set to the
-  built-in AWS provider callback.
+    {
+      configureFailPoint: "failCommand",
+      mode: {
+        times: 1
+      },
+      data: {
+        failCommands: [
+          "find"
+        ],
+        closeConnection: true
+      }
+    }
+
+- Perform a ``find`` operation that fails. This is to force the ``MongoClient``
+  to cache an access token.
 - Set a fail point for ``saslStart`` commands of the form:
```

Comment:
This doesn't raise a failure because it will successfully connect in speculative auth.

Reply:
Replaced this test with one that works with speculative auth.